Operational-Procedures/TSSW_Build/TSSW-Build-System.rst

```diff
@@ -0,0 +1,380 @@
+.. Do not remove other content provided in the templates, such as a section. Instead, comment out the content and include comments to explain the situation. For example:
+		- If a section within the template is not needed, comment out the section title and label reference. Include a comment explaining why this is not required.
+	- If a file cannot include a title (surrounded by ampersands (#)), comment out the title from the template and include a comment explaining why this is implemented (in addition to applying the ``title`` directive).
+
+.. Include one Primary Author and list of Contributors (comma separated) between the asterisks (*):
+.. |author| replace:: *Rob Bovill*
+.. If there are no contributors, write "none" between the asterisks. Do not remove the substitution.
+.. |contributors| replace:: *List-of-contributors*
+
+.. This is the label that can be used as for cross referencing this procedure.
+.. Recommended format is "Directory Name"-"Title Name"  -- Spaces should be replaced by hyphens.
+.. _Build-TSSW-Build-System:
+.. Each section should includes a label for cross referencing to a given area.
+.. Recommended format for all labels is "Title Name"-"Section Name" -- Spaces should be replaced by hyphens.
+.. To reference a label that isn't associated with an reST object such as a title or figure, you must include the link an explicit title using the syntax :ref:`link text <label-name>`.
+.. An error will alert you of identical labels during the build process.
+
+#################
+TSSW Build System
+#################
+
+.. _TSSW-Build-System-Introduction:
+
+Introduction
+^^^^^^^^^^^^
+
+This document will detail the setup, configuration and usage of the Telescope & Site Software (TSSW) Build System.  The build system is used to take the source code for an application and “build” it into a deployable artifact.  The build process can be very simple or very complex; it takes source code and any additional dependencies, especially software libraries, and then runs processes to transform the code into a deployable artifact.  This might be an RPM, a Conda package, an executable file, or some other packaging protocol.  A key piece of the build process is to run through the unit tests, before creating the package that will be deployed into a test or production environment.  If the unit tests fail, or fall below some tolerance threshold, then the build should terminate and not produce a package.
+
+The first section of this document will describe the TSSW build environment; which services and applications are used; what packaging methods were selected; and how the builds are coordinated.  The second section will define the build process and go into detail on how the build system integrates into the TSSW build and release process.  The third section will describe in detail the setup and configuration of the TSSW build environment.  This section will contain enough detail to replicate the build environment, if necessary.
```

Comment:
When you mention the sections, you could add links to them. This would help those users that may want to go directly there.

Reply:
fixed